pinot-core/src/main/java/org/apache/pinot/core/operator/filter/ScanBasedFilterOperator.java

```diff
@@ -48,10 +49,16 @@ public class ScanBasedFilterOperator extends BaseFilterOperator {
   @Override
   protected FilterBlock getNextBlock() {
     DataSourceMetadata dataSourceMetadata = _dataSource.getDataSourceMetadata();
+    ForwardIndexReader<?> forwardIndexReader = _dataSource.getForwardIndex();
```

Comment:
Instead of `getNextBlock`(), we can probably fail sooner in the constructor of `ScanBasedFilterOperator` which is still during physical planning ?

Reply:
Good catch! moved this to the constructor